src/database/Connector.java

```diff
@@ -0,0 +1,477 @@
+package database;
+import codeu.chat.common.Conversation;
+import codeu.chat.common.Message;
+import codeu.chat.common.Uuid;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.Timestamp;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.Scanner;
+import org.apache.commons.dbcp2.BasicDataSource;
+
+/**
+ * JDBC connector is needed to connects java program to the database.
+ * Description:
+ * this is the database connector you will need to use to communicate with
+ * the databased hosted on a remote machine.
+ */
+
+public class Connector {
```

Comment:
Overall this class is waaaaay cleaner than the first time I saw it, excellent progress.